server/repository/src/db_diesel/store_tag_row.rs

```diff
@@ -0,0 +1,248 @@
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct StoreTags {
+    pub store_id: String,
+    pub tags: Vec<String>,
+}
+
+pub struct StoreTagRowRepository<'a> {
+    connection: &'a StorageConnection,
+}
+
+impl<'a> StoreTagRowRepository<'a> {
+    pub fn new(connection: &'a StorageConnection) -> Self {
+        StoreTagRowRepository { connection }
+    }
+
+    pub fn insert_one(&self, store_tag_row: &StoreTagRow) -> Result<(), RepositoryError> {
+        diesel::insert_into(store_tag_dsl::store_tag)
+            .values(store_tag_row)
+            .execute(&self.connection.connection)?;
+        Ok(())
+    }
+
+    pub fn find_one_by_id(&self, id: &str) -> Result<Option<StoreTagRow>, RepositoryError> {
+        let result = store_tag_dsl::store_tag
+            .filter(store_tag_dsl::id.eq(id))
+            .first(&self.connection.connection)
+            .optional()?;
+        Ok(result)
+    }
+
+    pub fn find_all_by_store_id(
+        &self,
+        store_id: &str,
+    ) -> Result<Vec<StoreTagRow>, RepositoryError> {
+        let result = store_tag_dsl::store_tag
+            .filter(store_tag_dsl::store_id.eq(store_id))
+            .load(&self.connection.connection)?;
+        Ok(result)
+    }
+
+    pub fn find_all_by_tag_name(
+        &self,
+        tag_name: &str,
+    ) -> Result<Vec<StoreTagRow>, RepositoryError> {
+        let result = store_tag_dsl::store_tag
+            .filter(store_tag_dsl::tag_name.eq(tag_name))
+            .load(&self.connection.connection)?;
+        Ok(result)
+    }
+
+    pub fn find_one_by_store_id_and_tag_name(
+        &self,
+        store_id: &str,
+        tag_name: &str,
+    ) -> Result<Option<StoreTagRow>, RepositoryError> {
+        let result = store_tag_dsl::store_tag
+            .filter(store_tag_dsl::store_id.eq(store_id))
+            .filter(store_tag_dsl::tag_name.eq(tag_name))
+            .first(&self.connection.connection)
+            .optional()?;
+        Ok(result)
+    }
+
+    pub fn delete(&self, id: &str) -> Result<(), RepositoryError> {
+        diesel::delete(store_tag_dsl::store_tag.filter(store_tag_dsl::id.eq(id)))
+            .execute(&self.connection.connection)?;
+        Ok(())
+    }
+
+    pub fn delete_all_for_store(&self, store_id: &str) -> Result<(), RepositoryError> {
+        diesel::delete(store_tag_dsl::store_tag.filter(store_tag_dsl::store_id.eq(store_id)))
+            .execute(&self.connection.connection)?;
+        Ok(())
+    }
+
+    pub fn delete_many_by_id(&self, ids: Vec<String>) -> Result<(), RepositoryError> {
+        diesel::delete(store_tag_dsl::store_tag.filter(store_tag_dsl::id.eq_any(ids)))
+            .execute(&self.connection.connection)?;
+        Ok(())
+    }
+
+    pub fn upsert_tags_for_store(&self, new_tags: &StoreTags) -> Result<(), RepositoryError> {
+        // Find current tags for store
+        let current_tags = self.find_all_by_store_id(&new_tags.store_id)?;
```

Comment:
delete_all_for_store and insert &StoreTags ? Wondering why we would need the extra logic (do we really need to optimise number of db operations ?)